packages/sanity/src/tasks/src/tasks/hooks/useDocumentPreviewValues.ts

```diff
@@ -0,0 +1,45 @@
+import {ElementType, ReactNode} from 'react'
```

Comment:
Thanks for this hook, way cleaner the implementation like this.
Just thinking, will it be better to move this to `core` instead of tasks? As it's not using anything tasks specific and we could probably reuse it in other places.

Reply:
Long term I think we should introduce it into core, but I want us to live with it for a bit. We already have a few different preview related hooks that do similar things in core, and i'm not sure where this hook overlap with the others.